src/gsi/apply_scaledepwgts.f90

```diff
@@ -68,7 +68,7 @@ subroutine init_mult_spc_wgts(jcap_in)
   integer(i_kind) :: l_sum_spc_weights
 
   ! Spectral scale decomposition is differernt between SDL-cross and SDL-nocross
-  if( i_ensloccov4scl == 1 )then
+  if( r_ensloccov4scl == zero )then
```

Comment:
Here is one example when equality evaluation is used for the option in the real number.  As mentioned in another comment, this kind of use of real numbers are not safe.